ykcompile/src/lib.rs

```diff
@@ -125,9 +128,15 @@ impl TraceCompiler {
         }
     }
 
-    fn free_register(&mut self, local: &Local) {
+    fn free_register(&mut self, local: &Local) -> Result<(), CompileError> {
         if let Some(reg) = self.assigned_regs.remove(local) {
-            if local == &self.rtn_var.as_ref().unwrap().local {
+            if local
```

Comment:
Is this how `rustfmt` formats this? Can we not at least move the `== &self` into the line above, e.g.
```
            if local == &self
                       .rtn_var
                       .as_ref()
                       .ok_or_else(|| CompileError::NoReturnLocal)?
                       .local
```

Reply:
This is sadly how rustfmt formats this :P